node/src/main.rs

```diff
@@ -574,6 +572,29 @@ fn async_main() -> impl Future<Item = (), Error = ()> + Send + 'static {
             .expect("Failed to start GraphQL subscription server"),
     );
 
+    // Periodically check for contention in the tokio threadpool. First spawn a
+    // task that simply responds to "ping" requests. Then spawn a separate
+    // thread to periodically ping it and check responsiveness.
+    let (ping_send, ping_receive) = mpsc::channel::<std::sync::mpsc::Sender<()>>(1);
+    tokio::spawn(
+        ping_receive
+            .for_each(move |pong_send| pong_send.clone().send(()).map(|_| ()).map_err(|_| ())),
+    );
+    let contention_logger = logger.clone();
+    std::thread::spawn(move || loop {
+        std::thread::sleep(Duration::from_secs(10));
```

Comment:
My gut is telling me that 10 seconds is too sparse, contention could come and go during that window. 

This seems cheap enough to run every 10ms - 100ms. Is that the case?

Reply:
Agreed, I'll lower to 100ms.